### ISSUE TITLE
[BUG] ProviderStates is not generated when no pact version is defined

### ISSUE THREAD

**Issue by victorenaud** [pact-foundation/pact-jvm#1705]
Subjet: `au.com.dius.pact.consumer:junit5:4.6.1`

Given I have a test class anotated like this
```java
@PactTestFor(providerName = "ArticleAPI")
```

When setting a state with some parameters
```java
builder.given("an article exists", Map.of("id", "42")
...
```

Then only this is generated in my pact file:
```json
"providerState": "an article exists"
```

while the expected behavior is to have this generated:
```json
"providerStates": [
  {
    "name": "an article exists",
    "params": {
      "id": "42"
    }
  }
],
```

I believe the cause for that bug is in `RequestResponseInteraction.kt` at the line 32:
```java
    if (pactSpecVersion < PactSpecVersion.V3 && providerStates.isNotEmpty()) {
      interactionJson["providerState"] = providerStates.first().name.toString()
    } else if (providerStates.isNotEmpty()) {
      interactionJson["providerStates"] = providerStates.map { it.toMap() }
    }
    ```

As I've not defined the pact version in my test, it is set as `PactSpecVersion.UNSPECIFIED`, which in the above `if` is considered as _inferior_ to `PactSpectVersion.V3`, and so only the `providerState` is generated, and not the full `providerStates`.
Shouldn't this if be fixed to default `PactSpecVersion` to `V3` as it is done everywhere else?

**Comment by rholshausen** [comment]
Good point, it is probably time to set the default to V3. The reason it was not, is that there are lots of implementations that only support up to V2. But this is slowly changing.

**Comment by rholshausen** [comment]
Default is now V3, but can be overridden with the system property `pact.defaultVersion`
